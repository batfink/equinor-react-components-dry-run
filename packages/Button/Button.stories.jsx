import React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import Button from './lib/Button';
import equinorTheme from '../EquinorTheme/lib/EquinorTheme';
import defaultTheme from '../DefaultTheme/lib/DefaultTheme';
import { createGlobalStyle } from 'styled-components';
import EquinorLight from '../EquinorTheme/fonts/Equinor-Light.woff';
import EquinorRegular from '../EquinorTheme/fonts/Equinor-Regular.woff';
import EquinorMedium from '../EquinorTheme/fonts/Equinor-Medium.woff';
import EquinorBold from '../EquinorTheme/fonts/Equinor-Bold.woff';
import EquinorLightItalic from '../EquinorTheme/fonts/Equinor-LightItalic.woff';
import EquinorItalic from '../EquinorTheme/fonts/Equinor-Italic.woff';
import EquinorMediumItalic from '../EquinorTheme/fonts/Equinor-MediumItalic.woff';
import EquinorBoldItalic from '../EquinorTheme/fonts/Equinor-BoldItalic.woff';

const EquinorFonts = createGlobalStyle`
    @font-face {
        font-family: Equinor;
        src: local('☻'), url(${EquinorLight}) format('woff');
        font-style: normal;
        font-weight: 300;
    }
    @font-face {
        font-family: Equinor;
        src: local('☻'), url(${EquinorRegular}) format('woff');
        font-style: normal;
        font-weight: 400;
    }
    @font-face {
        font-family: Equinor;
        src: local('☻'), url(${EquinorMedium}) format('woff');
        font-style: normal;
        font-weight: 500;
    }
    @font-face {
        font-family: Equinor;
        src: local('☻'), url(${EquinorBold}) format('woff');
        font-style: normal;
        font-weight: 700;
    }
    @font-face {
        font-family: Equinor;
        src: local('☻'), url(${EquinorLightItalic}) format('woff');
        font-style: italic;
        font-weight: 300;
    }
    @font-face {
        font-family: Equinor;
        src: local('☻'), url(${EquinorItalic}) format('woff');
        font-style: italic;
        font-weight: 400;
    }
    @font-face {
        font-family: Equinor;
        src: local('☻'), url(${EquinorMediumItalic}) format('woff');
        font-style: italic;
        font-weight: 500;
    }
    @font-face {
        font-family: Equinor;
        src: local('☻'), url(${EquinorBoldItalic}) format('woff');
        font-style: italic;
        font-weight: 700;
    }
`;

const DefaultFont = createGlobalStyle`
    @import url('https://fonts.googleapis.com/css?family=Lato');
`;

const stories = storiesOf('Button', module);

const Wrapper = styled.div`
    display: grid;
    align-items: center;
    justify-items: center;
    height: 100vh;
    background: #ffffff;
`;

const ButtonWrapper = styled.div`
    display: flex;
    justify-content: space-around;
    width: 100%;
    max-width: 300px;
`;

stories.addDecorator(story => <Wrapper>{story()}</Wrapper>);

stories.add('Buttons with Equnior theme', () => (
    <ButtonWrapper>
        <EquinorFonts />
        <Button theme={equinorTheme} variant="primary">
            Primary
        </Button>
        <Button theme={equinorTheme} variant="default">
            Default
        </Button>
    </ButtonWrapper>
));

stories.add('Buttons with Default theme', () => (
    <ButtonWrapper>
        <DefaultFont />
        <Button theme={defaultTheme} variant="primary">
            Primary
        </Button>
        <Button theme={defaultTheme} variant="default">
            Default
        </Button>
    </ButtonWrapper>
));
