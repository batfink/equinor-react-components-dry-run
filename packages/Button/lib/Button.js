import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import defaultTheme, { themeProps } from '../../DefaultTheme/lib/DefaultTheme';

const CSS = (focusColor, hoverBackground) => ({
    primary: css`
        :hover {
            border-color: ${hoverBackground};
        }
        :focus {
            outline: 0 !important;
            border: 1px solid ${focusColor};
            box-shadow: inset 0 0 0 1px white, 0 0 0 1px ${focusColor};
        }
    `,
    default: css`
        :focus {
            border: 1px solid ${focusColor};
            box-shadow: 0 0 0 1px ${focusColor};
            transition: all 200ms cubic-bezier(0.165, 0.84, 0.44, 1);
        }
    `
});

const StyledButton = styled.button`
    font-family: ${({ fontFamily }) => fontFamily}, sans-serif;
    color: ${({ color }) => color};
    background: ${({ background }) => background};
    border-color: ${({ borderColor }) => borderColor};
    :hover {
        color: ${({ hoverColor }) => hoverColor};
        background: ${({ hoverBackground }) => hoverBackground};
    }
    text-decoration: none;
    font-size: 18px;
    border-radius: 5px;
    padding: 10px 20px;
    :disabled {
        cursor: not-allowed;
    }
    :focus {
        outline: none;
    }
    ${({ variant, focusColor, hoverBackground }) =>
        CSS(focusColor, hoverBackground)[variant]};
`;

const Button = ({
    theme = defaultTheme,
    variant = 'default',
    size = 'medium',
    ...attributes
}) => {
    const themeVariant = theme[variant] || {};
    const {
        color,
        background,
        borderColor,
        focusColor,
        hoverColor,
        hoverBackground
    } = themeVariant;
    const { fontFamily } = theme;
    return (
        <StyledButton
            size={size}
            variant={variant}
            fontFamily={fontFamily}
            color={color}
            background={background}
            borderColor={borderColor}
            focusColor={focusColor}
            hoverColor={hoverColor}
            hoverBackground={hoverBackground}
            {...attributes}
        />
    );
};

Button.propTypes = {
    theme: themeProps,
    variant: PropTypes.oneOf(['default', 'primary']),
    size: PropTypes.oneOf(['small', 'medium', 'large'])
};

export default Button;
