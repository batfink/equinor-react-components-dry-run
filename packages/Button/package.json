{
  "name": "button",
  "version": "0.0.1",
  "description": "Button html",
  "keywords": [
    "Button",
    "html"
  ],
  "author": "fortena <arnigeir@gmail.com>",
  "homepage": "https://github.com/batfink/equinor-react-components-dry-run#readme",
  "license": "ISC",
  "main": "lib/Button.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/batfink/equinor-react-components-dry-run.git"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1"
  },
  "bugs": {
    "url": "https://github.com/batfink/equinor-react-components-dry-run/issues"
  }
}
