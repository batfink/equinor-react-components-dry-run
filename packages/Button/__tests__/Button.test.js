import React from 'react';
import Button from '../lib/Button';
import renderer from 'react-test-renderer';

describe('Button', () => {
    it('should render a default Button', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a primary Button', () => {
        const tree = renderer.create(<Button varient="primary" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
