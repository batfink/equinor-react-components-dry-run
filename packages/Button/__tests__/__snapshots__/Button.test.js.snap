// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a default Button 1`] = `
.c0 {
  font-family: Lato,sans-serif,sans-serif;
  color: black;
  background: #f4f9fb;
  border-color: gray;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 18px;
  border-radius: 5px;
  padding: 10px 20px;
}

.c0:hover {
  color: black;
  background: white;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:focus {
  outline: none;
}

.c0:focus {
  border: 1px solid #68d2ff;
  box-shadow: 0 0 0 1px #68d2ff;
  -webkit-transition: all 200ms cubic-bezier(0.165,0.84,0.44,1);
  transition: all 200ms cubic-bezier(0.165,0.84,0.44,1);
}

<button
  className="c0"
  color="black"
  fontFamily="Lato, sans-serif"
  size="medium"
/>
`;

exports[`Button should render a primary Button 1`] = `
.c0 {
  font-family: Lato,sans-serif,sans-serif;
  color: black;
  background: #f4f9fb;
  border-color: gray;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 18px;
  border-radius: 5px;
  padding: 10px 20px;
}

.c0:hover {
  color: black;
  background: white;
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:focus {
  outline: none;
}

.c0:focus {
  border: 1px solid #68d2ff;
  box-shadow: 0 0 0 1px #68d2ff;
  -webkit-transition: all 200ms cubic-bezier(0.165,0.84,0.44,1);
  transition: all 200ms cubic-bezier(0.165,0.84,0.44,1);
}

<button
  className="c0"
  color="black"
  fontFamily="Lato, sans-serif"
  size="medium"
/>
`;
